### DISCUSSION TITLE
Can't publish `friendlyName` via MQTT

### DISCUSSION THREAD

**Discussion by guydvir2** [arendst/Tasmota#19567] [Q&A] [answered]
Hi, 
I'm trying to send MQTT msg to a global topic for all tasmota devices, each time Relay<x> has change. Since I have a 4 relay WallSwitch, I think the best way is to pub its friendly name and new state ( ON/OFF).

for example 
`backlog Rule2 ON Power1#State=1 DO Publish DvirHome/Messages [%timestamp%] [%topic%] [Relay1][ON] ENDON
ON Power1#State=0 DO Publish DvirHome/Messages [%timestamp%] [%topic%][Relay1] [OFF] ENDON
ON Power2#State=1 DO Publish DvirHome/Messages [%timestamp%] [%topic%][Relay2] [ON] ENDON
ON Power2#State=0 DO Publish DvirHome/Messages [%timestamp%] [%topic%][Relay2] [OFF] ENDON;
rule2 1`

when using %friendlyname1% instead `Relay1` - MQTT msg fails (not sent).
What can be done to publish it correctly?

Guy

**Comment by sfromis** [comment, marked as answer]
As there are no `%friendlyname1%` etc variables, what you try to do is not relevant. You have only these: https://tasmota.github.io/docs/Rules/#rule-variables

**Reply by sfromis** [reply]
If you want to use the FriendlyName values without repeating them, you could create rules to capture those in vars, which you then can use in the publish.

**Reply by guydvir2** [reply]
Adding rule as :`ON System#Boot DO VAR5 %friendlyName1% endon` saved "%friendlyName1%" to `Var5`.
What do I do wrong ?

**Reply by sfromis** [reply]
I **already** told you that
> there are no `%friendlyname1%` etc variables

"Extracting" via rules would be the command `Backlog FriendlyName` to get a JSON payload with values, and picking up data via a JSON-based [trigger](https://tasmota.github.io/docs/Rules/#rule-trigger) in another rule.

**Reply by guydvir2** [reply]
I got confused since 'friendlyName1' in console does yeild the corrent information I seek.
Got to admit that the first part, 'Backlog FriendlyName' is not fully understood.

**Reply by sfromis** [reply]
If the command does not provide what you think is "correct", better be clear about what you get, and why you think that it is wrong. And using `Backlog FriendlyName` is useful when you go on to create rules due to a technical quirk, but unless you do it from a rule, it makes no difference.

**Comment by guydvir2** [comment]
@sfromis  Sorry- I don't undestand your explanation.
The first part, it is not a variable - I get it.

The second part - I understand the idea of what you've said, but technically' I don't know how to implment it.
When typing in the console 'Backlog FriendlyName' - I get the  JSON with its 8 keys/ values. Now how to store it for a later use (JSON triggered), is still not understood.

I'll appreciate in example.
Guy

**Reply by barbudor** [reply]
```
Rule1
  On system#init do backlog FriendlyName endon
  On FriendlyName1 do var5 %value% endon
```

**Reply by guydvir2** [reply]
thank you. 
'Backlog FreindlyName' -  what actually does it mean ( I never used in such manner )
'On FriendlyName1 do'.... refers to a change in JSON ? ( again never used such trigger)

**Reply by sfromis** [reply]
I already posted a link explaining JSON triggers. They are very similar to the `#state` triggers, but are based on JSON already appearing in payloads for MQTT (even if not connected).

As mentioned, using `Backlog` in yjod context is due to a technical quirk, as JSON from commands issued by rules will generally not be picked up if you "just" issue the command (like `FriendlyName`) without "nesting" it in `Backlog`.
